### PR TITLE
Respect loaded mask cube properly in spectral extraction.

### LINKED ISSUE #3312: [BUG] Masks not handled properly in cube spectral extraction
### Jdaviz component

Cubeviz

### Description

While fixing #3307 I realized that we're not handling masks correctly at the moment for cubes. When we load a data cube, if it has a mask attribute that gets split out into a separate mask cube. Currently, that input mask is not being respected when we do a spectral extraction on the flux cube, leading to potentially incorrect extraction results and buggy behavior downstream when trying to, e.g., fit a model to the extracted spectrum. We need to handle input masks properly and apply them to both the flux and uncertainty cubes when extracting. 

### How to Reproduce

Load a MaNGA cube, attempt to fit a model to the full cube extracted spectrum, see that it fails (because all uncertainties are non-finite).

### Expected behavior

_No response_

### Browser

_No response_

### Jupyter

_No response_

### Software versions

_No response_

### PR DESCRIPTION
**Pull request by rosteen**
Fixes #3312.

Close https://github.com/spacetelescope/jdaviz/pull/2718

### CODE REVIEW

File: jdaviz/configs/cubeviz/helper.py
Comment (pllim) [inline jdaviz/configs/cubeviz/helper.py:25]: Maybe give @javerbukh co-author credit given https://github.com/spacetelescope/jdaviz/pull/2718 ?

**Review by kecnry — COMMENTED** [review]
The code itself looks good to me - just a few workflow related questions to consider:

Do we want (now or as a a follow-up ticket) to have a dropdown to select the mask cube?  Should the default be to apply the mask or to not apply it?

What happens if the intersection between an aperture and the mask results in no pixels?

Should the mask be used in other plugins as well (aperture photometry, moment map, collapse, etc)?

**Comment by camipacifici** [comment]
Visualize the mask cube would be very useful!

**Comment by gibsongreen** [comment]
Code seems straightforward, testing this on top of #3307 and the non-finite warning above the Fit Model button goes away with this PR.

In 3307, I was still seeing a model in the spectrum-viewer/table after model fitting, and there wasn't a traceback (but there was a warning was in the log). 

I'll hold off approving for now and re-review if/once the above comments are decided on  and if they go into this PR or follow-ups.

**Comment by rosteen** [comment]
@kecnry I just tested, if you define an aperture that is totally covered by the mask, it extracts a flat spectrum at 0 flux.

**Comment by rosteen** [comment]
I think I figured out the test failure - the science validation cube has NaNs in the mask/DQ array where the data isn't masked rather than 0s (btw, I did confirm that what we're storing from reading the mask array is what we're showing as the DQ data entry). This is awful, but easy to handle.

**Comment by rosteen** [comment]
I removed the lines storing the JWST DQ array as the original mask, and added a note in the docs as well as a Snackbar message indicating the original mask was applied during extraction, to address @camipacifici's concerns.

Screenshot 2024-12-05 at 1 11 40 PM

**Review by kecnry — APPROVED** [review]
Do we have any existing test data that we can send through spectral extraction to add coverage to these logic paths?  Otherwise looks good to me until we can get the full support in to allow mask selection across all relevant plugins.

**Review by gibsongreen — APPROVED** [review]
The recent code updates look good to me and behavior has been maintained since my last review!

**Pull request merged**
